source/io.c

```diff
@@ -808,44 +803,48 @@ int s_aws_input_stream_py_get_status(struct aws_input_stream *stream, struct aws
 int s_aws_input_stream_py_get_length(struct aws_input_stream *stream, int64_t *out_length) {
     (void)stream;
     (void)out_length;
-    return AWS_ERROR_UNIMPLEMENTED;
+    return aws_raise_error(AWS_ERROR_UNIMPLEMENTED);
+}
+
+struct aws_input_stream *s_aws_input_stream_py_acquire(struct aws_input_stream *stream) {
+    if (stream) {
+        /*************** GIL ACQUIRE ***************/
+        PyGILState_STATE state;
+        if (aws_py_gilstate_ensure(&state)) {
```

Comment:
optional optimization: instead of acquiring the GIL every time C refcount changes, we could only acquire the GIL and do Py_INCREF when refcount goes from 0->1, and vice-versa when C refcount goes from 1->0